From 362e789fe9d50bf3abc7d8b724767498562afd12 Mon Sep 17 00:00:00 2001
From: Peter Swain <pswain@cavium.com>
Date: Tue, 6 Feb 2018 15:30:11 -0800
Subject: [PATCH 0008/1051] mtd: nand: add Micron MT29F1G01AAADD, via
 staging/mt29f_spinand

Adds this 1Gb SPI part with internal ECC.
Still exploring ECC warning:
  nand: device found, Manufacturer ID: 0x2c, Chip ID: 0x12
  nand: Micron NAND 128MiB 3,3V 1-bit
  nand: 128 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
  nand: WARNING: spi0.1: the ECC used on your system is too weak compared to the one required by the NAND chip

Signed-off-by: Peter Swain <pswain@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/mtd/nand/raw/nand_ids.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/drivers/mtd/nand/raw/nand_ids.c b/drivers/mtd/nand/raw/nand_ids.c
index 5423c3bb388e..f604a4db8ae5 100644
--- a/drivers/mtd/nand/raw/nand_ids.c
+++ b/drivers/mtd/nand/raw/nand_ids.c
@@ -164,6 +164,20 @@ struct nand_flash_dev nand_flash_ids[] = {
 	EXTENDED_ID_NAND("NAND 64GiB 1,8V 16-bit", 0x2E, 65536, LP_OPTIONS16),
 	EXTENDED_ID_NAND("NAND 64GiB 3,3V 16-bit", 0x4E, 65536, LP_OPTIONS16),
 
+	/* Micron MT29F1G01AAADD, via staging/mt29f_spinand */
+	{
+		.name = "NAND 128MiB 3,3V 1-bit",
+		.mfr_id = NAND_MFR_MICRON,
+		.dev_id = 0x12,
+		.pagesize = 2048,
+		.chipsize = 128,
+		.erasesize = SZ_128K,
+		.options = NAND_BBT_SCAN2NDPAGE | NAND_ECC_CUSTOM_PAGE_ACCESS,
+		.id_len = 2,
+		.oobsize = 64,
+		.ecc = NAND_ECC_INFO(4, 512),
+	},
+
 	{NULL}
 };
 
-- 
2.17.1

